pubsub-adapter/src/main/kotlin/com/ably/pubsub/RealtimeChannel.kt

```diff
@@ -0,0 +1,175 @@
+   * on the channel are received by any channel listeners registered using [subscribe].
+   * Any resulting channel state change will be emitted to any listeners registered using the
+   * [io.ably.lib.util.EventEmitter.on] or [io.ably.lib.util.EventEmitter.once] methods.
+   * As a convenience, `attach()` is called implicitly if [subscribe] for the channel is called,
+   * or [RealtimePresence.enter] or [RealtimePresence.subscribe] are called on the [RealtimePresence] object for this channel.
+   *
+   * Spec: RTL4d
+   */
+  fun attach(listener: CompletionListener? = null)
+
+  /**
+   * Detach from this channel.
+   * Any resulting channel state change is emitted to any listeners registered using the
+   * [io.ably.lib.util.EventEmitter.on] or [io.ably.lib.util.EventEmitter.once] methods.
+   * Once all clients globally have detached from the channel, the channel will be released in the Ably service within two minutes.
+   *
+   * Spec: RTL5e
+   */
+  fun detach(listener: CompletionListener? = null)
+
+  /**
+   * Registers a listener for messages on this channel.
+   * The caller supplies a listener function, which is called each time one or more messages arrives on the channel.
+   *
+   * Spec: RTL7a
+   *
+   * @param listener A listener may optionally be passed in to this call to be notified of success or failure
+   * of the channel [RealtimeChannel.attach] operation. This listener is invoked on a background thread.
+   */
+  fun subscribe(listener: MessageListener)
+
+  /**
+   * Registers a listener for messages with a given event name on this channel.
+   * The caller supplies a listener function, which is called each time one or more matching messages arrives at the channel.
+   *
+   * Spec: RTL7b
+   *
+   * @param eventName The event name.
+   * @param listener A listener may optionally be passed in to this call to be notified of success or failure
+   * of the channel [RealtimeChannel.attach] operation. This listener is invoked on a background thread.
+   */
+  fun subscribe(eventName: String, listener: MessageListener)
+
+  /**
+   * Registers a listener for messages on this channel for multiple event name values.
+   * The caller supplies a listener function, which is called each time one or more matching messages arrives on the channel.
+   *
+   * Spec: RTL7a
+   *
+   * @param eventNames A list of event names.
+   * @param listener A listener may optionally be passed in to this call to be notified of success or failure
+   * of the channel [RealtimeChannel.attach] operation. This listener is invoked on a background thread.
+   */
+  fun subscribe(eventNames: List<String>, listener: MessageListener)
+
+  /**
+   * Deregisters the given listener (for any/all event names).
+   * This removes an earlier subscription.
+   *
+   * Spec: RTL8a
+   *
+   * @param listener An event listener function.
+   */
+  fun unsubscribe(listener: MessageListener)
```

Comment:
Is this wrapper perhaps an opportunity to revisit how we do `unsubscribe`, e.g. how we do it in chat?

Reply:
Agreed, fixed